### ISSUE TITLE
Remove-Migration doesn't support Connection, but requires it

### ISSUE THREAD

**Issue by coreyfournier** [dotnet/efcore#24271]
### Is your feature request related to a problem? Please describe.
A clear and concise description of what the problem is.
I have multiple projects, but a single .net core library contains the DbContext and migration files. This all works fine where I am able to specify the Context, Project, and Connection. Except for "Remove-Migration". It doesn't allow the parameter Connection. Because of this I have to default a connection in the DbContext.

### Describe the solution you'd like
I would like to be able to specify the Connection like all other functions support in the Remove-Migration function

**Comment by jikuja** [comment]
Because #25252 was marked as duplicate of this:

`dotnet ef migrations remove` help documentation should be improved: 

**Revert the migration if it has been applied to the database.** does not give any hint to user that argument is also used to ignore connection related issues.

**Comment by lauxjpn** [comment]
I second this issue. This always catches me cold once in a while when I just want to remove a migration and suddenly my database is being hit.

The _default_ for `dotnet ef migrations remove` needs to not hit any database. An optional parameter to also change the database could be considered, but technically this is already covered by the `dotnet ef database update` command.

Generally, I think that only the `dotnet ef database` commands should be able to change a database.

**Comment by chrisfcarroll** [comment]
This is currently awful :-( 

Sanity imho would be one of:
- accept and document a `--connection` parameter like `migrations list` does.
- rename the `--force` parameter to `--offline` and document it as such

**Comment by ska737** [comment]
> The _default_ for `dotnet ef migrations remove` needs to not hit any database.

This has gotten me a lot. I'm setting up the migrations before applying them (I'm migrating a project and am just trying to get the table definitions to match, not actually creating the database). I don't want to create the database just so I can remove the last migration that was not correct.

> Generally, I think that only the `dotnet ef database` commands should be able to change a database.

This makes total sense. The only things that should even attempt to hit a database are the `dotnet ef database` commands. Everything else should be segregated.

**Comment by Erumer** [comment]
I completely agree: having `migrations` command act on code, and `database` on db would be much more clear

**Comment by stormbreakermerika** [comment]
Yep this would be nice to know.  Just ran into this situation and had to go looking for ` dotnet ef migrations remove --force`

Upvote from me on this issue

**Comment by mdsharpe** [comment]
Shocked that this has yet to be addressed. `dotnet ef migrations remove` is just broken now imo.

**Comment by roji** [comment]
@mdsharpe many people are using it quite successfully, and this issue has only received 17 votes in the last 4 years, which is why we haven't prioritized it.

**Comment by LunicLynx** [comment]
I think the amount of people that voted for this is not a good metric. It requires to understand that you can't just delete the migrations first. And I have seen people doing this the source control way rather then using this command. This functionality should honestly just be there.
